drivers/clock_control/clock_control_tt_bh.c

```diff
@@ -344,13 +344,13 @@ static void clock_control_tt_bh_update(const struct clock_control_tt_bh_config *
 	/* Setup external postdivs */
 	clock_control_tt_bh_config_ext_postdivs(config, settings);
 
-	k_busy_wait(300);
+	k_busy_wait(0.3);
```

Comment:
I'm surprised that this compiles without a warning, since it's an implicit cast from `float` to `uint32_t`.

Bringup code does include a `WaitNs()` function. If we can avoid using it, that would be nice.

Zephyr lacks a `k_busy_wait_ns()` function currently though, likely because few  Zephyr devices would need it.

Can you add this zephyr patch file as a workaround until `k_busy_wait_ns()` has a proper implementation?
```diff
diff --git a/include/zephyr/kernel.h b/include/zephyr/kernel.h
index 614b2eab1b9..e9752bb0fdb 100644
--- a/include/zephyr/kernel.h
+++ b/include/zephyr/kernel.h
@@ -617,6 +617,11 @@ __syscall int32_t k_usleep(int32_t us);
  */
 __syscall void k_busy_wait(uint32_t usec_to_wait);
 
+static inline void k_busy_wait_ns(uint32_t nsec_to_wait)
+{
+       k_busy_wait(k_cyc_to_us_ceil32(k_ns_to_cyc_floor32(nsec_to_wait)));
+}
+
 /**
  * @brief Check whether it is possible to yield in the current context.
  *
```

```suggestion
	k_busy_wait_ns(300);
```